src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,326 @@
+        {
+            var automationElement = await LocateAutomationElementAsync(elementName, recursive).ConfigureAwait(continueOnCapturedContext: false);
+
+            object invokePattern = null;
+            if (automationElement.TryGetCurrentPattern(InvokePattern.Pattern, out invokePattern))
+            {
+                ((InvokePattern)(invokePattern)).Invoke();
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            // Explicitly don't call GC.SuppressFinalize() so that the finalizer is still called and the _hostProcess is killed
+        }
+
+        public void Initialize()
+        {
+            Cleanup();
+
+            if (RequireNewInstance)
+            {
+                Debug.WriteLine("Starting a new instance of Visual Studio.");
+
+                InitializeHostProcess();
+                InitializeDte();
+                InitializeRemotingService();
+            }
+        }
+
+        internal async Task ExecuteDteCommandAsync(string command, string args = "")
+        {
+            await WaitForDteCommandAvailabilityAsync(command).ConfigureAwait(continueOnCapturedContext: false);
+            _dte.ExecuteCommand(command, args);
+        }
+
+        internal T ExecuteOnHostProcess<T>(Type type, string methodName, BindingFlags bindingFlags, params object[] parameters)
+             => ExecuteOnHostProcess<T>(type.Assembly.Location, type.FullName, methodName, bindingFlags, parameters);
+
+        internal T ExecuteOnHostProcess<T>(string assemblyFilePath, string typeFullName, string methodName, BindingFlags bindingFlags, params object[] parameters)
+        {
+            var objectUri = _service.Execute(assemblyFilePath, typeFullName, methodName, bindingFlags, parameters);
+
+            if (string.IsNullOrWhiteSpace(objectUri))
+            {
+                return default(T);
+            }
+
+            return (T)(Activator.GetObject(typeof(T), $"{_serviceUri}/{objectUri}"));
+        }
+
+        internal async Task<AutomationElement> LocateAutomationElementAsync(string elementName, bool recursive = false)
+        {
+            AutomationElement automationElement = null;
+            var scope = (recursive ? TreeScope.Descendants : TreeScope.Children);
+            var condition = new PropertyCondition(AutomationElement.NameProperty, elementName);
+
+            await IntegrationHelper.WaitForResultAsync(() => {
+                automationElement = AutomationElement.RootElement.FindFirst(scope, condition);
+                return (automationElement != null);
+            }, expectedResult: true).ConfigureAwait(continueOnCapturedContext: false);
+
+            return automationElement;
+        }
+
+        internal Window LocateDteWindow(string windowTitle)
+            => LocateDteWindowAsync(windowTitle).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
```

Comment:
ConfigureAwait has no meaning here. Also, again sync-atop-async is not recommended practice.